de.metas.adempiere.adempiere/base/src/main/java-legacy/org/compiere/model/MLookupFactory.java

```diff
@@ -690,7 +690,14 @@ private static String buildDisplayColumnSql(final ILookupDisplayColumn ldc, fina
 		// date
 		else if (DisplayType.isDate(ldc.getDisplayType()))
 		{
-			return DB.TO_CHAR(columnSQL, ldc.getDisplayType());
+			
```

Comment:
i think you shall do that only if DisplayType.Date.
For all others keep as it was.
NOTE: DIsplayType.isDate is returning true for Date, DateTIme and Time